onnx/optimizer/optimize.h

```diff
@@ -14,25 +15,20 @@
 
 namespace ONNX_NAMESPACE { namespace optimization {
 
-void PrepareOutput(const ONNX_NAMESPACE::ModelProto& mp_in, ONNX_NAMESPACE::ModelProto& mp_out);
+ONNX_NAMESPACE::ModelProto PrepareOutput(const ONNX_NAMESPACE::ModelProto& mp_in);
 
-struct Optimizer {
+struct Optimizer final {
```

Comment:
If user want, they can have their own optimizer, which is a sub-class of our Optimizer class.

Reply:
Ok makes sense. I removed `final` and added a virtual destructor to avoid partial destruction when people subclass it.